### PR TITLE
Reuse Template Service

### PR DESCRIPTION
**Pull request by nicoburniske**
- Reuse TemplateCompilationService
- Reuse Template WASM AST processing/transformation

Spent too much time trying to create an abstraction for TemplateRepo, and ultimately decided it's not worth it. We are supporting Postgres and SQLite in OSS and it got too complicated

### CODE REVIEW

File: golem-template-service/src/config.rs [hunk 1]
```diff
@@ -114,7 +99,6 @@ impl Default for TemplateServiceConfig {
             db: DbConfig::default(),
             template_store: TemplateStoreConfig::default(),
             compilation: TemplateCompilationConfig::default(),
-            worker_executor_client_cache: WorkerExecutorClientCacheConfig::default(),
```
Comment (nicoburniske) [inline golem-template-service/src/config.rs:117]: I am not sure why we had this. it's not used anywhere

**Review by justcoon — APPROVED** [review]

File: golem-template-service/src/config.rs (hunk 1 above)
Comment (afsalthaj) [reply golem-template-service/src/config.rs:117]: GRPC client cache for worker comms from template.  Its confusing we are not using this, something to consider

File: golem-template-service-base/src/service/mod.rs [hunk 2]
```diff
@@ -0,0 +1,2 @@
+pub mod template_compilation;
+pub mod template_processor;
```
Comment (afsalthaj) [inline golem-template-service-base/src/service/mod.rs:2]: Interesting that reusable components here is far less than I expected.

File: golem-template-service/src/config.rs (hunk 1 above)
Comment (vigoo) [reply golem-template-service/src/config.rs:117]: The template service is not calling worker executor at all, right?

Comment (afsalthaj) [reply golem-template-service/src/config.rs:117]: Oh okay, Ok it got not used after we have a separate worker-service that talks to worker-executor.

**Review by afsalthaj — APPROVED** [review]

**Comment by afsalthaj** [comment]
Its unfortunate the data layer isn't reusable. It's unrelated to the PR and it was a good decision to solve it in this PR. Inmy view, its because some of suboptimal storage we did in the beginning. Things such as `Project` `account` etc overly tied to the core entities.

**Pull request merged**